Storage/tests/Unit/Connection/RestTest.php

```diff
@@ -273,6 +277,92 @@ function ($args) use (&$actualRequest, $response) {
         );
     }
 
+    public function downloadObjectWithResumeProvider()
+    {
+        $part1 = 'first part';
+        $part2 = 'second part';
+        $full = $part1 . $part2;
+        /** @see RetryTrait::$httpRetryCodes for list of status codes that are allowed to retry */
+        return [
+            [0, '', 200, $full, $full, []],
+            [200, $part1, 200, $part2, $full, ['Range' => 'bytes=10-']],
+            [408, 'Server Message', 200, $full, $full, []],
+            [429, 'Server Message', 200, $full, $full, []],
+            [500, 'Server Error', 200, $full, $full, []],
+            [502, 'Server Error', 200, $full, $full, []],
+            [503, 'Server Error', 200, $full, $full, []],
+            [504, 'Server Error', 200, $full, $full, []],
+        ];
+    }
+
+    /**
+     * @dataProvider downloadObjectWithResumeProvider
+     */
+    public function testDownloadObjectWithResume(int $status1,
+                                                 string $body1,
+                                                 int $status2,
+                                                 string $body2,
+                                                 string $expectedResult,
+                                                 array $expectedSecondRequestHeaders)
+    {
```

Comment:
nit, this indentation is a bit rough to me
```suggestion
    public function testDownloadObjectWithResume(
        int $status1,
        string $body1,
        int $status2,
        string $body2,
        string $expectedResult,
        array $expectedSecondRequestHeaders
    ) {
```